spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/JobUtil.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.bootstrap.data.util;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AsyncJobStatusDTO;
+import org.candlepin.resource.JobsApi;
+
+public final class JobUtil {
+
+    /** The default amount of time we should wait for a job to finish processing **/
+    private static final long DEFAULT_TIMEOUT_MILLISECONDS = 60000;
+
+    private JobUtil() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Waits for a given job to finish processing based on the provided job Id.
+     *
+     * @param jobId - Id of the job to wait for.
+     * @param api - api used to determine if job has completed.
+     * @return true if the job has finished processing successfully or false if the job
+     *         has failed or exceeded the default timeout duration.
+     * @throws ApiException
+     * @throws InterruptedException
+     */
+    public static boolean waitForJobToComplete(String jobId, JobsApi api)
+        throws ApiException, InterruptedException {
+        return waitForJobToComplete(jobId, DEFAULT_TIMEOUT_MILLISECONDS, api);
+    }
+
+    /**
+     * Waits for a given job to finish processing based on the provided job Id.
+     *
+     * @param jobId - Id of the job to wait for.
+     * @param timeout - max duration in milliseconds to wait for the job to finish.
+     * @param api - api used to determine if job has completed.
+     * @return true if the job has finished processing successfully or false if the job
+     *         has failed or exceeded the provided timeout duration.
+     * @throws ApiException
+     * @throws InterruptedException
+     */
+    public static boolean waitForJobToComplete(String jobId, long timeout, JobsApi api)
```

Comment:
Returning a boolean is a bit cumbersome here. When using this method we want to be able to assert on AsyncJobStatusDTO in general (the status field mostly, but also the rest), so it would be better if this method returns the AsyncJobStatusDTO itself. And the while loop below stops when either we've entered one of the [4 terminal states](https://github.com/candlepin/candlepin/blob/bb8d37d50c11d1eaa92751e8dabd741bc420d38e/src/main/java/org/candlepin/model/AsyncJobStatus.java#L77-L83), or the timeout has elapsed

Reply:
Yeah, definitely makes more sense to return the AsyncJobStatusDTO.